src/Tests/Microsoft.NET.Build.Tests/GivenThatWeHaveAPackageReferenceWithAliases.cs

```diff
@@ -0,0 +1,115 @@
+            testProject.PackageReferences.Add(
+                new TestPackageReference(
+                    packageReferences.Last().ID,
+                    packageReferences.Last().Version,
+                    packageReferences.Last().NupkgPath,
+                    packageReferences.Last().PrivateAssets,
+                    aliases: "Special"));
+            var packagesPaths = packageReferences.Select(e => Path.GetDirectoryName(e.NupkgPath));
+
+            testProject.AdditionalProperties.Add("RestoreAdditionalProjectSources",
+                                     "$(RestoreAdditionalProjectSources);" + string.Join(";", packagesPaths));
+
+            //  Use a test-specific packages folder
+            testProject.AdditionalProperties["RestorePackagesPath"] = @"$(MSBuildProjectDirectory)\..\pkg";
+            testProject.SourceFiles[$"{testProject.Name}.cs"] = ClassLibUsage;
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var buildCommand = new BuildCommand(Log, Path.Combine(testAsset.TestRoot, testProject.Name));
+            buildCommand.Execute()
+                .Should()
+                .Pass();
+        }
+
+        private IEnumerable<TestPackageReference> GetPackageReferencesWithConflictingTypes(string targetFramework, params string[] packageNames)
+        {
+            foreach (var packageName in packageNames)
+            {
+                yield return GetPackageReferenceWithConflictingType(targetFramework, packageName);
+            }
+        }
+
+        private TestPackageReference GetPackageReferenceWithConflictingType(string targetFramework, string packageName)
+        {
+            var project = GetProjectWithConflictingTypes(targetFramework, packageName);
+            var packCommand = new PackCommand(Log, _testAssetsManager.CreateTestProject(project).TestRoot, packageName);
+
+            packCommand
+                .Execute()
+                .Should()
+                .Pass();
+            return new TestPackageReference(packageName, "1.0.0", packCommand.GetNuGetPackage(packageName));
+        }
+
+        private static TestProject GetProjectWithConflictingTypes(string targetFramework, string referenceProjectName)
+        {
+            var project = new TestProject()
+            {
+                Name = referenceProjectName,
+                TargetFrameworks = targetFramework,
+                IsSdkProject = true
+            };
+            project.SourceFiles[$"{referenceProjectName}.cs"] = ClassLib;
+            return project;
+        }
+
+        private static string ClassLib = @"
+using System;
+public class ClassLib
+{
+    public void ConflictingMethod()
+    {
+    }
+}
+";
+
+        private static string ClassLibUsage = @"
+extern alias Special;
+using System;
+public class ClassLibUsage
+{
+    public void UsageMethod()
+    {
+        new Special.ClassLib().ConflictingMethod();
```

Comment:
Can we update the test, or add a test, to cover the case where you are calling multiple classes with the same name?  I think you'd need two aliases for that.

Reply:
Also I think it would be nice to cover the case where there are multiple aliases for a single assembly, if that's a supported scenario.

Reply:
I can add tests for both of those scenarios.

Reply:
Added more tests.